File: JavaScript_12.18/3159 - Tijolão.js
/**
 * @author Josélio de S. C. Júnior <joseliojrx25@gmail.com>
 * @copyright Josélio de S. C. Júnior 2021
*/

const l = require('fs').readFileSync('/dev/stdin', 'utf8')
  .replace(/\r/g, '')
  .split('\n');
const c = [];
const key = 'abc def ghi jkl mno pqrs tuv wxyz'.split(' ');
let n = 0;
key.forEach((e, i) =>
  e.split('').forEach((_, j) => {
    c[String.fromCharCode(97 + n)] = `${i + 2}`.repeat(j + 1);
    n++;
  })
);
Array.from(Array(+l.shift()), _ => {
  const str = l.shift()
    .split('')
    .map(e => c[e] ? c[e]
      : e === ' ' ? '0'
      : `#${c[e.toLowerCase()]}`)
    .map((e, i, a) => {
      if (a[i + 1] && a[i + 1].match(/^\d/))
        return e.match(/\d$/)[0] === a[i + 1].match(/^\d/)[0] ? `${e}*` : e;
      return e;
    })
    .join('');
  console.log(str);
});

File: Python_3.9/3159 - Tijolão.py
# author: Josélio Júnior <joseliojrx25@gmail.com>
# copyright: Josélio Júnior 2022

k = [
  'abc', 'def', 'ghi',
  'jkl', 'mno', 'pqrs',
  'tuv', 'wxyz'
]

for _ in range(int(input())):
  rw = []
  r = input()

  for l in r:
    if l == ' ':
      rw.append('0')

    if l.isupper():
      rw.append('#')
      l = l.lower()

    i = 2
    for e in k:
      if l in e:
        si = e.index(l) + 1
        rw.append(str(i) * si)
      i += 1

  res = []
  if len(rw) > 1:
    for i in range(len(rw) - 1):
      if rw[i][-1] == rw[i + 1][0]:
        res.append(rw[i] + '*')
      else:
        res.append(rw[i])
    res.append(rw[-1])
  else:
    res = rw
  print(''.join(res))
